src/System.Linq.Parallel/src/System/Linq/ParallelEnumerable.cs

```diff
@@ -5354,18 +5351,20 @@ public static ParallelQuery<TResult> OfType<TResult>(this ParallelQuery source)
         }
 
         /// <summary>
-        /// Converts the elements of a ParallelQuery to the specified type.
+        /// Converts the elements of a weakly-typed ParallelQuery to the specified stronger type.
         /// </summary>
-        /// <typeparam name="TResult">The type to convert the elements of <paramref name="source"/> to.</typeparam>
-        /// <param name="source">The sequence that contains the elements to be converted.</param>
+        /// <typeparam name="TResult">The stronger type to convert the elements of <paramref name="source"/> to.</typeparam>
+        /// <param name="source">The sequence that contains the weakly typed elements to be converted.</param>
         /// <returns>
-        /// A sequence that contains each element of the source sequence converted to the specified type.
+        /// A sequence that contains each weakly-type element of the source sequence converted to the specified stronger type.
         /// </returns>
         /// <exception cref="T:System.ArgumentNullException">
         /// <paramref name="source"/> is a null reference (Nothing in Visual Basic).
         /// </exception>
         public static ParallelQuery<TResult> Cast<TResult>(this ParallelQuery source)
         {
+            if (source == null) throw new ArgumentNullException("source");
```

Comment:
While this makes sense and is arguably how it should have been in the first place, it's also a behavioral change from the desktop.  @AlfredoMS, is this ok?